### PR TITLE
CATL-1020: Load Default Value For A Contact From Case Relationships

### PR DESCRIPTION
**Pull request by vinuvarshith**
Overview
----------------------------------------
This feature is already existing for non case relationships. We want to load the default value of the existing contact field on a webform through a case relationship
scrn

Before
----------------------------------------
setup of the webform to test what's existing:
When we go to the webform - just as an example - we can add 2 contacts
Contact 1 can be the current user
We can set contact 2's default value to load from a relationship. This can be done by editing the existing contact field of Contact 2 on the webform, and setting the 'default value' to load from a relationship.
See screenshot.
This, however, only sets about non case relationships. There's currently no way to load a contact from case relationships (case roles)

After
----------------------------------------
In the first dropdown where we specify the source of loading (Set default contact from) - we currently have 4 options: None, Specified Contact, Relationship To, Auto - from filters

Now there is another option to this dropdown: 'Case Relationship to'
The second field is to select a contact (remains the same)
The third field where we need to specify the relationship - on selecting 'Case Relationship to' - this will now load only case relationships (case roles)
This option will only be available when cases in enabled on the webform.

In addition if only the case id is given in url and if 'contact 1: existing contact' item in webform config is set to load 'defaults' from 'case roles' and 'case client' is set for contact 1, then this will automatically load the first case client of the case passed in the url. (Previously passing only the case would not autofill anything)
Also 'Current User' option is now available for all contacts (Previously it was only available for contact1)

### CODE REVIEW

**Comment by shitijg** [comment]
@jitendrapurohit guess you asked for this?

**Comment by shitijg** [comment]
@colemanw if you can review this one - that will be great.

**Comment by jitendrapurohit** [comment]
I've applied this PR on my review setup. I'll post the test result soon.

File: includes/contact_component.inc [hunk 1]
```diff
@@ -238,6 +240,31 @@ function _webform_edit_civicrm_contact($component) {
       '#parents' => array('extra', 'default_relationship'),
     );
   }
+  
+  $case_count = $data['case']['number_of_case'];
```
Comment (jitendrapurohit) [inline includes/contact_component.inc]: I get a notice error on this line while editing a contact component - 

    Notice: Undefined index: case in _webform_edit_civicrm_contact() (line 244 of /Applications/MAMP/htdocs/civicrm-master/sites/all/modules/webform_civicrm/includes/contact_component.inc).

**Review by jitendrapurohit — COMMENTED** [review]
Looks pretty much great. This does fix the requirement of #171 and much more flexibility for case contacts. I've found a single notice error as of now. Can you pls push an update to fix that?

Thanks for the PR.

**Comment by vinuvarshith** [comment]
@jitendrapurohit The above notice is fixed. Thanks.

**Comment by shitijg** [comment]
@colemanw can we merge this?

**Pull request merged**